libs/gl-cli/src/node.rs

```diff
@@ -0,0 +1,327 @@
+    let creds = match util::read_credentials(&creds_path) {
+        Some(c) => c,
+        None => {
+            return Err(Error::CredentialsNotFoundError(format!(
+                "could not read from {}",
+                creds_path.display()
+            )))
+        }
+    };
+
+    let scheduler = gl_client::scheduler::Scheduler::new(config.network, creds)
+        .await
+        .map_err(Error::custom)?;
+
+    let mut node: gl_client::node::Client = scheduler.node().await.map_err(Error::custom)?;
+    let mut stream = node
+        .stream_log(StreamLogRequest {})
+        .await
+        .map_err(Error::custom)?
+        .into_inner();
+
+    //TODO:  Use tokio::select!() and ctrl + c to cancel listening for log lines
+    loop {
+        tokio::select! {
+            biased;
+            _ = tokio::signal::ctrl_c() => {
+                break;
+            }
+            maybe_line = stream.next() => {
+                match maybe_line {
+                    Some(line) => {
+                        println!("{}", line.map_err(Error::custom)?.line);
+                    },
+                    None => {
+                        break;
+                    },
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+async fn stop<P: AsRef<Path>>(config: Config<P>) -> Result<()> {
+    let creds_path = config.data_dir.as_ref().join(CREDENTIALS_FILE_NAME);
+    let creds = match util::read_credentials(&creds_path) {
+        Some(c) => c,
+        None => {
+            return Err(Error::CredentialsNotFoundError(format!(
+                "could not read from {}",
+                creds_path.display()
+            )))
+        }
+    };
+
+    let scheduler = gl_client::scheduler::Scheduler::new(config.network, creds)
+        .await
+        .map_err(Error::custom)?;
+
+    let mut node: gl_client::node::ClnClient = scheduler.node().await.map_err(Error::custom)?;
+    let res = node
+        .stop(cln::StopRequest {})
+        .await
+        .map_err(|e| Error::custom(e.message()))?
+        .into_inner();
+    println!("{:?}", res);
+    Ok(())
+}
+
+async fn getinfo_handler<P: AsRef<Path>>(config: Config<P>) -> Result<()> {
+    let creds_path = config.data_dir.as_ref().join(CREDENTIALS_FILE_NAME);
+    let creds = match util::read_credentials(&creds_path) {
+        Some(c) => c,
+        None => {
+            return Err(Error::CredentialsNotFoundError(format!(
+                "could not read from {}",
+                creds_path.display()
+            )))
+        }
+    };
+
+    let scheduler = gl_client::scheduler::Scheduler::new(config.network, creds)
+        .await
+        .map_err(Error::custom)?;
+
+    let mut node: gl_client::node::ClnClient = scheduler.node().await.map_err(Error::custom)?;
+    let res = node
+        .getinfo(cln::GetinfoRequest {})
+        .await
+        .map_err(|e| Error::custom(e.message()))?
+        .into_inner();
+    println!("{:?}", res);
```

Comment:
I think longer term we'll want a JSON conversion for these. Do you think it'd be possible to instrument `prost` in such a way that we generate the JSON equivalent output?

Reply:
That is a good question. I think this should be possible the problem here is more that `cln_rpc` and `cln_grpc` use several models (or messages) where the representation of public keys, features and other stuff is in bytes. We would need to add custom deserialization for the `Vec<u8>` fields into e.g. hex strings. Maybe the better place for this would be the upstream libs.

Reply:
I think we could likely go to cln-rpc and generate the hex conversion annotations in the generated code too 🤔